tests/ci/integration/run_openvpn_integration.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+set -exu
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+
+# SYS_ROOT
+#  - SRC_ROOT(aws-lc)
+#    - SCRATCH_FOLDER
+#      - OPENVPN_SRC_FOLDER
+#      - AWS_LC_BUILD_FOLDER
+#      - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER="${SRC_ROOT}/OPENVPN_BUILD_ROOT"
+OPENVPN_SRC_FOLDER="${SCRATCH_FOLDER}/openvpn"
+OPENVPN_BUILD_PREFIX="${OPENVPN_SRC_FOLDER}/build/install"
+OPENVPN_BUILD_EPREFIX="${OPENVPN_SRC_FOLDER}/build/exec-install"
+OPENVPN_PATCH_BUILD_FOLDER="${SRC_ROOT}/tests/ci/integration/openvpn_patch"
+
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf "${SCRATCH_FOLDER:?}"/*
+cd ${SCRATCH_FOLDER}
+
+function openvpn_build() {
+  autoreconf -ivf
+
+  OPENSSL_CFLAGS="-I/${AWS_LC_INSTALL_FOLDER}/include" \
+  OPENSSL_LIBS="-L/${AWS_LC_INSTALL_FOLDER}/lib -lssl -lcrypto" \
+  ./configure \
+    --prefix="$OPENVPN_BUILD_PREFIX" \
+    --exec-prefix="$OPENVPN_BUILD_EPREFIX" \
+    --with-crypto-library=openssl \
+    --with-openssl-engine=no \
+    --disable-management
+
+  make
+  make install
+
+  export LD_LIBRARY_PATH="${AWS_LC_INSTALL_FOLDER}/lib"
+
+  local openvpn_executable="${OPENVPN_SRC_FOLDER}/build/exec-install/sbin/openvpn"
+  ldd ${openvpn_executable} \
+    | grep "${AWS_LC_INSTALL_FOLDER}/lib/libcrypto.so" || exit 1
+}
+
+# TODO: Remove this when we make an upstream contribution.
+function openvpn_patch_build() {
+  for patchfile in $(find -L "${OPENVPN_PATCH_BUILD_FOLDER}" -type f -name '*.patch'); do
+    echo "Apply patch $patchfile..."
+    patch -p1 --quiet -i "$patchfile"
+  done
+}
+
+function openvpn_run_tests() {
+  # Explicitly running as sudo and passing in LD_LIBRARY_PATH as some OpenVPN
+  # tests run as sudo and LD_LIBRARY_PATH doesn't get inherited.
+  sudo LD_LIBRARY_PATH="${AWS_LC_INSTALL_FOLDER}/lib" make check
```

Comment:
why does this require root permissions?

Reply:
One of the integration tests (t_net.sh) uses sudo to modify iptables and conduct tests. The LD_LIBRARY_PATH set in the un-privileged environment does not get inherited by root user when this test is executed. Therefore, we need to explicitly pass in LD_LIBRARY_PATH as root user when running the tests to ensure the correct libcrypto is being used. 

Test link: https://github.com/OpenVPN/openvpn/blob/856065b2eb37286c389550593472bf180bc5be9d/tests/t_net.sh#L54

Reply:
makes sense for GitHub CI where integration tests are run in isolation and everything is ephemeral. i'm a bit worried about idempotency when running these on dev machines. today, we [don't use `sudo`](https://github.com/search?q=repo%3Aaws%2Faws-lc+path%3Atests%2Fci%2Fintegration%2F*+sudo&type=code) for integration tests. i hope openvpn is configurable enough be able to run these tests on configurable (i.e. unprivileged) ports. if not, there are [ways](https://github.com/WillChilds-Klein/pq-tls-benchmark/blob/master/emulation-exp/code/kex/setup.sh#L47) to reasonably guard against network state snafus.

let's sync up on this offline, i'm happy to work through it with you :)

Reply:
mm, reading through t_net.sh a bit more, it looks like they're careful enough to clean up state and don't `set -e` so early exit shouldn't be an issue. i think we're good here.